rfc-089-removing-the-government-prefix-route.md

```diff
@@ -0,0 +1,54 @@
+# Removing the /government prefix route
+
+## Summary
+
+The /government prefix route is a large source of uncertainty, and
+because of this uncertainty it is quite difficult to work out what
+specific content still uses it. Changing the way the traffic is
+handled could offer a methodical way of tracking what the remaining
+work is to make the /government prefix route redundant.
+
+## Problem
+
+The /government prefix route is a large source of uncertainty in the
+context of the rest of the routing on GOV.UK. It's hard to tell what
+it's responsible for, and thus what work needs doing to get Whitehall
+content to manage it's own routes.
+
+Also, while I think it would be possible in the router to have
+separate handlers for /governement as an exact route, and /government
+as a prefix route, this isn't supported in either the Publishing API
+or Router API. This means that /government (the exact path) is
+currently stuck being rendered by whitehall-frontend.
+
+## Proposal
+
+There are some known groups of content rendered by whitehall-frontend
+for which the routing information isn't explicitly set in the
+Publishing API.
+
+However, even if that were fixed, due to the nature of the problem,
+the uncertainty of what this prefix route is responsible for, it would
+be hard to tell what routing information is missing from what content.
+
+### Action Plan
```

Comment:
I wonder if we can do this too by getting a months worth of requests from the CDN logs, filtering by `/government` prefix and checking if they have exact matches in the router. That would save us from having to alter the infrastructure. What do you think?

Reply:
> I wonder if we can do this too by getting a months worth of requests from the CDN logs, filtering by /government prefix and checking if they have exact matches in the router. That would save us from having to alter the infrastructure. What do you think?

I think that approach would work, however if putting in the effort of writing something to check for the existence of routes is worth it, it would be more useful to check against the routes in the Publishing API, as that would give better data (as it would avoid routes that have crept in to the router outside of the Publishing API).

Overall, while doing the route checking in code rather than using the running router service would avoid changing the infrastructure, my feeling is that it would take more effort to setup (although using data from the Publishing API could have an added benefit).

Reply:
> my feeling is that it would take more effort to setup 

I think we could loop through the URLs and query the router directly? There would be no new code to write.

https://github.com/alphagov/router-api#getting-details-of-a-route

Reply:
> I think we could loop through the URLs and query the router directly? There would be no new code to write.

I was including that loop in the code that would need to be written and that considering prefix routes under /government as a "must have". Adding more specific prefix routes might be a practical way of getting rid of the /government prefix route and by my count there are quite a few prefix routes within /government already [1] that may be handling some requests.

1: 
```
irb(main):014:0> Route.where(route_type: "prefix", incoming_path: { '$regex': /^\/government\/.*/ }).count
=> 733
```